Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/PlatformInit.c

```diff
@@ -0,0 +1,67 @@
+/** @file
+  Platform initialization PEIM for QEMU
+
+  Copyright (c) 2022 Théo Jehl All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include "PlatformInit.h"
+#include <Library/PeimEntryPoint.h>
+#include <Library/PeiServicesLib.h>
+#include "Library/DebugLib.h"
+#include <Library/PlatformInitLib.h>
+#include <Library/HobLib.h>
+#include <Library/PciCf8Lib.h>
+#include <IndustryStandard/Pci.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/PcdLib.h>
+#include <IndustryStandard/Q35MchIch9.h>
+
+EFI_STATUS
+EFIAPI
+PlatformInit (
+  IN       EFI_PEI_FILE_HANDLE  FileHandle,
+  IN CONST EFI_PEI_SERVICES     **PeiServices
+  )
+{
+  EFI_STATUS             Status;
+  UINT16                 DeviceId;
+  EFI_HOB_PLATFORM_INFO  *EfiPlatformInfo;
+
+  // Install permanent memory
+  Status = InstallMemory (PeiServices);
+
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "Memory installation failed\n"));

```

Comment:
you can use %r and pass the status code to print the error messages associated with error codes